src/MCStates.jl

```diff
@@ -18,151 +19,118 @@ Fieldnames:
 - `beta`: inverse temperature
 - `config`: actual configuration in Markov chain [`Config`](@ref)  
 - `dist_2mat`: matrix of squared distances d_ij between atoms i and j; generated automatically when potential `pot` given
-- `en_atom_vec`: vector of energy contributions per atom i; generated automatically when `pot` given
+-`new_dist2_vec`: calculates the new r2 between atoms based on a trial move
+- `new_en` : new energy value for trial configuraiton
 - `en_tot`: total energy of `config`; generated automatically when `pot` given
+
+- `potential_variables` : mutable struct containing energy-related variables for the current configuration
+- `ensemble_variables` : mutable struct containing ensemble-related variables for the current configuraiton
 - `ham`: vector containing sampled energies - generated in MC run
-- `max_displ`: max_diplacements for atom, volume and rotational moves; key-word argument
 - `count_atom`: number of accepted atom moves - total and between adjustment of step sizes; key-word argument
 - `count_vol`: number of accepted volume moves - total and between adjustment of step sizes; key-word argument
-- `count_rot`: number of accepted rotational moves - total and between adjustment of step sizes; key-word argument
+
 - `count_exc`: number of attempted (10%) and accepted exchanges with neighbouring trajectories; key-word argument
 """
-mutable struct MCState{T,N,BC}
+mutable struct MCState{T,N,BC,PVType,EVType}
     temp::T
     beta::T
     config::Config{N,BC,T}
     dist2_mat::Matrix{T}
+    new_dist2_vec::Vector{T}
 
-    tan_mat::Matrix{T}
-    en_atom_vec::Vector{T}
-
+    new_en::T
     en_tot::T
+
+    potential_variables::PVType
+    ensemble_variables::EVType
+
     ham::Vector{T}
     max_displ::Vector{T}
     max_boxlength::T
     count_atom::Vector{Int}
     count_vol::Vector{Int}
-    count_rot::Vector{Int}
     count_exc::Vector{Int}
 end    
 
-function MCState(
-    temp, beta, config::Config{N,BC,T}, dist2_mat, tan_mat, en_atom_vec, en_tot; 
-    max_displ = [0.1,0.1,1.], max_boxlength = 10., count_atom = [0,0], count_vol = [0,0], count_rot = [0,0], count_exc = [0,0]
-) where {T,N,BC}
-    ham = T[]
-    MCState{T,N,BC}(
-        temp, beta, deepcopy(config), copy(dist2_mat), copy(tan_mat),copy(en_atom_vec), en_tot, 
-        ham, copy(max_displ), copy(max_boxlength), copy(count_atom), copy(count_vol), copy(count_rot), copy(count_exc)
-        )
+function max_length(bc::SphericalBC)
```

Comment:
??? what is the purpose of this function; no docu

Reply:
Not sure, not removing it is a case of Chesterton's fence.